src/app/ui/alpha_slider.h

```diff
@@ -0,0 +1,44 @@
+// Aseprite
+// Copyright (C) 2024  Igara Studio S.A.
+//
+// This program is distributed under the terms of
+// the End-User License Agreement for Aseprite.
+
+#ifndef APP_UI_ALPHA_SLIDER_H_INCLUDED
+#define APP_UI_ALPHA_SLIDER_H_INCLUDED
+#pragma once
+
+#include "app/pref/preferences.h"
+#include "ui/slider.h"
+
+namespace app {
+
+class AlphaSlider : public ui::Slider,
+                    ui::SliderDelegate
+{
+public:
+  enum Type {ALPHA, OPACITY};
+
+  AlphaSlider(int value, Type type)
+    : ui::Slider(0, 255, value, this)
+    , m_type(type) {}
+  void getSliderThemeInfo(int* min, int* max, int* value) const override;
+  void updateValue(int value);
+  app::gen::AlphaRange getAlphaRange() const {
+    return (m_type == ALPHA ? Preferences::instance().range.alpha()
+                            : Preferences::instance().range.opacity());
+  }
+
+protected:
+  bool onProcessMessage(ui::Message* msg) override;
+
+private:
+  Type m_type;
+  // ui::SliderDelegate impl
+  std::string onGetTextFromValue(int value) override;
+  int onGetValueFromText(const std::string& text) override;
+};
+
```

Comment:
Minor formatting changes can be made (+ moving AlphaSlider ctor to the .cpp file):
```suggestion
class AlphaSlider : public ui::Slider,
                    ui::SliderDelegate
{
public:
  enum Type { ALPHA, OPACITY };

  AlphaSlider(int value, Type type);
      
  void getSliderThemeInfo(int* min, int* max, int* value) const override;
  void updateValue(int value);
  
  app::gen::AlphaRange getAlphaRange() const {
    return (m_type == ALPHA ? Preferences::instance().range.alpha()
                            : Preferences::instance().range.opacity());
  }

protected:
  bool onProcessMessage(ui::Message* msg) override;

private:
  // ui::SliderDelegate impl
  std::string onGetTextFromValue(int value) override;
  int onGetValueFromText(const std::string& text) override;

  Type m_type;
};

```
Generally we want to put data members as the last items in the class definition (doesn't modify the structure of the class, but just for readability we want to put all member functions before data).